submissions/AsyaYeromina/Tiny-JS-world-OOP/index.js

```diff
@@ -0,0 +1,167 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+  Complete the below for code reviewers' convenience:
+
+  Code repository: https://github.com/AsyaYeromina/a-tiny-JS-world
+  Web app: https://asyayeromina.github.io/a-tiny-JS-world/.
+  */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Inhabitant {
+  constructor(inhabitantProperties) {
+    this.species = inhabitantProperties.species;
+    this.name = inhabitantProperties.name;
+    this.gender = inhabitantProperties.gender;
+    this.legs = inhabitantProperties.legs;
+    this.speech = inhabitantProperties.speech;
+    this.friends = inhabitantProperties.friends;
+  }
+}
+
+class Animal extends Inhabitant {
+  constructor(inhabitantProperties) {
+    super({ ...inhabitantProperties, legs: 4});
+    this.tail = inhabitantProperties.tail;
+  }
+
+  toString() {
+    return [
+      this.species,
+      this.name,
+      this.gender,
+      this.legs,
+      this.tail,
+      this.speech,
+      this.friends,
+    ]
+    .map((propertyValue) =>
+    propertyValue == this.tail && propertyValue === true
+      ? (propertyValue = "has tail")
+      : propertyValue == this.tail && propertyValue === false
+      ? (propertyValue = "no tail")
+      : propertyValue
```

Comment:
What happens here:

Issue 1.
`propertyValue == this.tail && propertyValue === true`  -- if either of properties (`this.species`, `this.name` etc including `this.tail`) has the same value as `this.tail` and the value is `true`, then "has tail" is a result from evaluation (`propertyValue` gets assigned but it is never stored in fact, see below for details).
So if it happens we've got a property e.g. `this.hasHorn` and it is `true` and we also have `this.tail`, which is also `true` then we will receive "has tail" twice.

What you most likely meant here is that if property **name** is `tail`, then... etc.

To make the latter logic to work you'd rather need property **names** as strings in the initial array. Then you'd be able to compare each value in the array against `"tail"` and build relevant logic. And whatever property name is not `"tail"` you then extract its value by means of `this[propName]` (i.e. `this["name"]` is equivalent of `this.name`)

Issue 2. Ternary
Whatever `propertyValue` is assigned within this ternary is never used (this variable, or rather callback argument, is scoped to this callback only). `propertyValue` in the subsequent `map` is a completely different variable, just with a similar name.

So, why ternary still returns "has tail" and "no tail"? Because expressions in JS **always** return something. So statement `propertyValue = "has tail"` returns "has tail". Try removing `propertyValue =` after both of `?` and note that the outcome doesn't change (parentheses also become obsolete while do not block any expession within).

Basically, this ternary here returns either of "has tail", "no tail" or a value received from the initial array.

Just of curiosity try executing each statements below, one by one in browser console and spot the output from each  of the statements.
```
let a = "a", b = "b";

a = "a", b = "b" // yes, it is comma and it is generally valid

(propertyValue = "has tail")

[1,2,3].forEach(n => n + 1)

[1,2,3].map(n => n + 1)
```